src/routing_node.rs

```diff
@@ -65,19 +67,21 @@ static MAX_BOOTSTRAP_CONNECTIONS : usize = 1;
 /// Routing Node
 pub struct RoutingNode {
     // for CRUST
-    crust_receiver      : mpsc::Receiver<crust::Event>,
-    connection_manager  : crust::ConnectionManager,
-    accepting_on        : Vec<crust::Endpoint>,
-    bootstraps          : BTreeMap<Endpoint, Option<NameType>>,
+    crust_receiver       : mpsc::Receiver<crust::Event>,
+    connection_manager   : crust::ConnectionManager,
+    accepting_on         : Vec<crust::Endpoint>,
+    bootstraps           : BTreeMap<Endpoint, Option<NameType>>,
     // for RoutingNode
-    action_sender       : mpsc::Sender<Action>,
-    action_receiver     : mpsc::Receiver<Action>,
-    event_sender        : mpsc::Sender<Event>,
-    filter              : MessageFilter<types::FilterType>,
-    core                : RoutingCore,
-    public_id_cache     : LruCache<NameType, PublicId>,
-    connection_cache    : BTreeMap<NameType, SteadyTime>,
-    accumulator         : MessageAccumulator,
+    action_sender        : mpsc::Sender<Action>,
+    action_receiver      : mpsc::Receiver<Action>,
+    event_sender         : mpsc::Sender<Event>,
+    filter               : MessageFilter<types::FilterType>,
+    core                 : RoutingCore,
+    public_id_cache      : LruCache<NameType, PublicId>,
+    connection_cache     : BTreeMap<NameType, SteadyTime>,
+    accumulator          : MessageAccumulator,
+    immutable_data_cache : LruCache<NameType, Data>,
```

Comment:
I could use ImmutableData. The only drawback would be converting it to the Data to send it in handle_cache_get.
I can change if you think ImmutableData is better